app/scripts/modules/core/src/pipeline/config/triggers/pipelineRoles.component.ts

```diff
@@ -0,0 +1,48 @@
+import { IController, module } from 'angular';
+import { AuthenticationService } from 'core/authentication';
+
+class PipelineRolesController implements IController {
+  public pipeline: any;
+  public allowedRoles: string[];
+
+  public $onInit(): void {
+    this.allowedRoles = AuthenticationService.getAuthenticatedUser().roles;
+    this.pipeline.roles = this.pipeline.roles || [];
+  }
+}
+
+class PipelineRolesComponent implements ng.IComponentOptions {
```

Comment:
Ideally everything would be pipeline "permissions" or pipeline "roles" - but I am ok with that happening in a different PR.